python/cudf/cudf/core/column_accessor.py

```diff
@@ -243,13 +222,6 @@ def _grouped_data(self) -> abc.MutableMapping:
         else:
             return self._data
 
-    @cached_property
```

Comment:
This caching had meaningful performance impacts in some contexts IIRC. Can we make `nrows` a cached property now to get the same performance? That would also require adding corresponding logic to `_clear_cache`.

Reply:
Sure thing. Added `cached_property` to `nrows`